File: JavaScript/leetcode_409.js
var longestPalindrome = function(s) {
    let counts = new Map();
    for (let char of s) {
        counts.set(char, (counts.get(char) || 0) + 1);
    }

    let length = 0;
    let oddFound = false;

    for (let count of counts.values()) {
        if (count % 2 === 0) {
            length += count;
        } else {
            length += count - 1;
            oddFound = true;
        }
    }

    if (oddFound) length += 1;

    return length;
};

File: Python/leetcode_409.py
from collections import Counter


def longest_palindrome(s: str) -> int:
    counts = Counter(s)
    length = 0
    odd_found = False

    for count in counts.values():
        if count % 2 == 0:
            length += count
        else:
            length += count - 1
            odd_found = True

    if odd_found:
        length += 1  # Add one odd character in the center

    return length
